### PR TITLE
Add validation for javascript source maps and source code

### PR DESCRIPTION
**Pull request by academo**
Adds a validation to check:

- The provided plugin archive contains the javascript source maps
- The sourcemaps in the archive match the provided source code
- These validations work with nested plugins

# Note for the reviewer

don't be alarmed by the amount of files. most of those files are test files necessary to test the functionality.

Do not review files inside `testdata` all the javascript and typescript there are merely fixtures.

### CODE REVIEW

File: pkg/analysis/passes/jssourcemap/jssourcemap.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
+package jssourcemap
```
Comment (academo) [inline pkg/analysis/passes/jssourcemap/jssourcemap.go:1]: Review this file

File: pkg/analysis/passes/manifest/manifest_test.go [hunk 2]
```diff
@@ -76,11 +76,7 @@ func TestManifestWithMissingFiles(t *testing.T) {
 		"File img/clock-extra.svg is not declared in MANIFEST.txt",
```
Comment (academo) [inline pkg/analysis/passes/manifest/manifest_test.go:76]: Review this file

File: pkg/difftool/difftool.go [hunk 3]
```diff
@@ -26,14 +26,20 @@ type sourceDiff struct {
 	Equal                    bool
```
Comment (academo) [inline pkg/difftool/difftool.go:26]: Review this file. Note this file was moved from `/cmd` to here.

File: pkg/testpassinterceptor/testpassinterceptor.go [hunk 4]
```diff
@@ -11,3 +11,18 @@ func (t *TestPassInterceptor) ReportInterceptor() func(string, analysis.Diagnost
 		t.Diagnostics = append(t.Diagnostics, &diagnostic)
 	}
 }
+
```
Comment (academo) [inline pkg/testpassinterceptor/testpassinterceptor.go:14]: Review this file

**Review by briangann — APPROVED** [review]
lgtm, ran and detected changes between source code and release zip as expected!

**Review by xnyo — APPROVED** [review]
LGTM great work!!

**Pull request merged**